library/src/main/java/com/karumi/headerrecyclerview/HeaderRecyclerViewAdapter.java

```diff
@@ -59,10 +59,26 @@
   }
 
   /**
+   * If you don't need header feature, you can bypass overriding this method.
+   */
+  protected VH onCreateHeaderViewHolder(ViewGroup parent, int viewType) {
+    return null;
+  }
+
+  protected abstract VH onCreateItemViewHolder(ViewGroup parent, int viewType);
+
+  /**
+   * If you don't need footer feature, you can bypass overriding this method.
+   */
+  protected VH onCreateFooterViewHolder(ViewGroup parent, int viewType) {
+    return null;
+  }
+
+  /**
```

Comment:
Header and footer are optional, so don't use `abstract` modifier to force user to override `onCreateHeaderViewHolder` and `onCreateFooterViewHolder`